src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlSessionCacheFactory.java

```diff
@@ -0,0 +1,101 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.stargate.sgv2.jsonapi.config.OperationsConfig;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaCache;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * Factory for creating a singleton {@link CQLSessionCache} instance that is configured via CDI.
+ *
+ * <p>We use this factory so the cache itself is not a CDI bean. For one so it does not have all
+ * that extra overhead, and because there construction is a bit more complicated.
+ */
+@ApplicationScoped
+public class CqlSessionCacheFactory {
+
+  private CQLSessionCache singleton = null;
```

Comment:
I am not sure I understand why the whole lazy construction logic here -- seems like all configuration is passed via constructor and construction of the cache could be made eager and access simpler?

Adding comments on why (assuming there are solid reasons -- some cyclic dependency) would be good. Or just make construction eager.